src/main/java/com/google/javascript/clutz/DeclarationGenerator.java

```diff
@@ -2586,7 +2525,13 @@ private void visitObjectType(
           ((ObjectType) instanceType).getOwnPropertyNames(),
           superClassFields,
           classTemplateTypeNames);
-      // Statics are handled in INSTANCE_CLASS_SUFFIX class.
+
+      // Interface statics were emitted in an sibling namespace prior to this point, because
```

Comment:
this needs more docs, it describes a situation, but not why the code below has to run.

Also, this method would benefit from breaking it up into sub steps, this is a pretty long read.

Reply:
Improved the comment. Will refactor later, as this PR is already on the large side.